cirq-google/cirq_google/circuit_serializer.py

```diff
@@ -0,0 +1,657 @@
+                gate.phase_exponent,
+                out=msg.phasedxpowgate.phase_exponent,
+                arg_function_language=arg_function_language,
+            )
+            arg_func_langs.float_arg_to_proto(
+                gate.exponent,
+                out=msg.phasedxpowgate.exponent,
+                arg_function_language=arg_function_language,
+            )
+            return msg
+        if isinstance(gate, cirq.PhasedXZGate):
+            arg_func_langs.float_arg_to_proto(
+                gate.x_exponent,
+                out=msg.phasedxzgate.x_exponent,
+                arg_function_language=arg_function_language,
+            )
+            arg_func_langs.float_arg_to_proto(
+                gate.z_exponent,
+                out=msg.phasedxzgate.z_exponent,
+                arg_function_language=arg_function_language,
+            )
+            arg_func_langs.float_arg_to_proto(
+                gate.axis_phase_exponent,
+                out=msg.phasedxzgate.axis_phase_exponent,
+                arg_function_language=arg_function_language,
+            )
+            return msg
+        if isinstance(gate, cirq.CZPowGate):
+            arg_func_langs.float_arg_to_proto(
+                gate.exponent,
+                out=msg.czpowgate.exponent,
+                arg_function_language=arg_function_language,
+            )
+            return msg
+        if isinstance(gate, cirq.ISwapPowGate):
+            arg_func_langs.float_arg_to_proto(
+                gate.exponent,
+                out=msg.iswappowgate.exponent,
+                arg_function_language=arg_function_language,
+            )
+            return msg
+        if isinstance(gate, cirq.FSimGate):
+            arg_func_langs.float_arg_to_proto(
+                gate.theta,
+                out=msg.fsimgate.theta,
+                arg_function_language=arg_function_language,
+            )
+            arg_func_langs.float_arg_to_proto(
+                gate.phi,
+                out=msg.fsimgate.phi,
+                arg_function_language=arg_function_language,
+            )
+            return msg
+        if isinstance(gate, cirq.MeasurementGate):
+            arg_func_langs.arg_to_proto(
+                gate.key,
+                out=msg.measurementgate.key,
+                arg_function_language=arg_function_language,
+            )
+            arg_func_langs.arg_to_proto(
+                gate.invert_mask,
+                out=msg.measurementgate.invert_mask,
+                arg_function_language=arg_function_language,
+            )
+            return msg
+        if isinstance(gate, cirq.WaitGate):
+            arg_func_langs.float_arg_to_proto(
+                gate.duration.total_nanos(),
+                out=msg.waitgate.duration_nanos,
+                arg_function_language=arg_function_language,
+            )
+            return msg
+        raise ValueError(f'Cannot serialize op {op!r} of type {type(gate)}')
```

Comment:
I would suggest changing this to a chain of `elif` with a final `else` that raises the exception, then having a single return at the end. Could also delay adding qubits and tags until after this check of gate types, since it doesn't make sense to do either of those if the gate type itself is not supported.

Reply:
Done.